pkg/task/azure.go

```diff
@@ -0,0 +1,63 @@
+package task
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/puppetlabs/relay-sdk-go/pkg/model"
+	"github.com/puppetlabs/relay-sdk-go/pkg/taskutil"
+	ini "gopkg.in/ini.v1"
+)
+
+func (ti *TaskInterface) ProcessAzure(directory string) error {
+	var spec model.AzureSpec
+
+	if err := taskutil.PopulateSpecFromDefaultPlan(&spec, ti.opts); err != nil {
+		return err
+	}
+
+	if spec.Azure == nil {
+		return nil
+	}
+
+	if directory == "" {
+		directory = filepath.Join(DefaultPath, ".azure")
+	}
+
+	// .azure/credentials
+	creds := ini.Empty()
+	creds.Section(spec.Azure.Connection.SubscriptionID).Key("client_id").SetValue(spec.Azure.Connection.ClientID)
+	creds.Section(spec.Azure.Connection.SubscriptionID).Key("secret").SetValue(spec.Azure.Connection.Secret)
+	creds.Section(spec.Azure.Connection.SubscriptionID).Key("tenant").SetValue(spec.Azure.Connection.TenantID)
```

Comment:
I think we want all four of `subscription_id`, `client_id`, `secret`, and `tenant` under a "profile" section. The azure cli will look for a `default` profile by default, but other profiles can have arbitrary names.

Maybe you have docs that describe using the subscription ID as the section name? I couldn't find a mention of that yet.

Reply:
Oh, I do see https://dev.to/azure/setting-up-the-azure-credentials-file-41o3 now... looking for a more definitive source

Reply:
Not even right; the format with `[default]` is for `ansible` not `az`. Whoops.

Reply:
Yeah, that's what I based it off of. Testing it out.

Reply:
That information was either outdated or incorrect...